docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,811 @@
+
+Once each client receives the ready messages from the two other nodes, it can
+immediately choose the correct fee per vByte from the provided values (see
+<<bitcoin-fee>>), construct the sighash (see <<sighash-construction>>) and
+proceed into the signing process to sign that sighash. Once a signature is
+computed, each client should construct the final transaction (see
+<<final-transaction-construction>>) and submit it (see
+<<transaction-submission>>).
+
+When receiving the ready message from other nodes, a node should validate that
+the announced BTC address can be parsed correctly by the tooling used to handle
+transaction and sighash construction. If the announced BTC address cannot be
+parsed, an error should be logged and the message should be treated as if it was
+not received.
+
+The retry mechanism for the signing process for liquidation recovery should be
+identical to regular signing, repeating every 2 minutes that a signature has
+not been completed successfully until the liquidation recovery timeout. Unlike
+standard signing, the liquidation recovery timeout should be very long to allow
+for node malfunctions to recover cleanly: this RFC recommends a default timeout
+of 7 days.
+
+[[transaction-submission]]
+==== Submitting the Signed Transaction
+
+Submitting a signed Bitcoin transaction requires a connection to a Bitcoin
+node, direct or indirect. To avoid _strictly requiring_ a connection to a
+Bitcoin node, this RFC proposes logging the signed transaction 5 times on the
+WARN level with a descriptive prefix. Current operator guidance suggests that
+more than 5 WARN level logs within 5 minutes should trigger a notification
+for closer monitoring by operators, ensuring these logs will be noticed. The
+descriptive prefix will further allow for targeted notifications by operators
+so that they can flag these logs for follow-up action.
+
+The suggested structure of the log message is:
+
+```
+2020-02-01T12:12:12.12-0800    WARN   keep-ecdsa   Please broadcast Bitcoin transaction 0010010101000828181124af234gbawed000
+```
+
+The first three components are the standard components emitted by the ECDSA
+client's logging library, while the message itself begins with `Please broadcast
+Bitcoin transaction`, followed by the hexadecimal string representing the signed
+transaction. The signed transaction can be broadcast via manual submission to a
+known node, via API access to a service provider like BlockCypher or
+Blockstream, or via a manual form submission on a page like
+https://live.blockcypher.com/btc/pushtx/ .
+
+If an optional Bitcoin node connection is available (see <<electrs>>), the
+client should attempt to publish the transaction by performing an HTTP POST
+request to path `/tx`. The POST request body should be the signed transaction
+hex, as would be logged in the WARN logs above. If the POST fails, the client
+should log an error and then fall back on the WARN logging above.
```

Comment:
We fall back to the WARN logging so that the signed transaction is available for publishing, rather than to call operator attention. I think it's simpler to consider it “the fallback transaction publishing path”, which _happens_ to be WARN logging. To that end, I think it's simpler to always have that path be the same, irrespective of what else is going on. I may be able to rephrase to make that clearer though.

Reply:
Rephrased this a bit.